test/e2e-go/restAPI/restClient_test.go

```diff
@@ -1228,3 +1230,299 @@ func TestNilStateProofInParticipationInfo(t *testing.T) {
+			a.True(createdBoxName[string(b.Name)])
+		}
+	}
+
+	testingBoxNames := []string{
+		` `,
+		`     	`,
+		` ? = % ;`,
+		`; DROP *;`,
+		`OR 1 = 1;`,
+		`"      ;  SELECT * FROM kvstore; DROP acctrounds; `,
+		`背负青天而莫之夭阏者，而后乃今将图南。`,
+		`於浩歌狂熱之際中寒﹔於天上看見深淵。`,
+		`於一切眼中看見無所有﹔於無所希望中得救。`,
+		`有一遊魂，化為長蛇，口有毒牙。`,
+		`不以嚙人，自嚙其身，終以殞顛。`,
+		`那些智力超常的人啊`,
+		`认为已经，熟悉了云和闪电的脾气`,
+		`就不再迷惑，就不必了解自己，世界和他人`,
+		`每天只管，被微风吹拂，与猛虎谈情`,
+		`他们从来，不需要楼梯，只有窗口`,
+		`把一切交付于梦境，和优美的浪潮`,
+		`在这颗行星所有的酒馆，青春自由似乎理所应得`,
+		`面向涣散的未来，只唱情歌，看不到坦克`,
+		`在科学和啤酒都不能安抚的夜晚`,
+		`他们丢失了四季，惶惑之行开始`,
+		`这颗行星所有的酒馆，无法听到远方的呼喊`,
+		`野心勃勃的灯火，瞬间吞没黑暗的脸庞`,
+		`b64:APj/AA==`,
+		`str:123.3/aa\\0`,
+		string([]byte{0, 255, 254, 254}),
+		string([]byte{0, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF}),
+		`; SELECT key from kvstore WHERE key LIKE %;`,
+		`?&%!=`,
+		"SELECT * FROM kvstore " + string([]byte{0, 0}) + " WHERE key LIKE %; ",
+		string([]byte{'%', 'a', 'b', 'c', 0, 0, '%', 'a', '!'}),
+		`
+`,
+		`™£´´∂ƒ∂ƒßƒ©∑®ƒß∂†¬∆`,
+		`∑´´˙©˚¬∆ßåƒ√¬`,
+	}
+
+	resp, err := testClient.ApplicationBoxes(uint64(createdAppID), 0)
+	a.NoError(err)
+	a.Empty(resp.Boxes)
+
+	for i := 0; i < len(testingBoxNames); i += 16 {
+		var strSliceTest []string
+		// grouping box names to operate, and create such boxes
+		if i+16 >= len(testingBoxNames) {
+			strSliceTest = testingBoxNames[i:]
+		} else {
+			strSliceTest = testingBoxNames[i : i+16]
+		}
+		operateAndMatchRes("create", strSliceTest)
+	}
+
+	maxBoxNumToGet := uint64(10)
+	resp, err = testClient.ApplicationBoxes(uint64(createdAppID), maxBoxNumToGet)
+	a.NoError(err)
+	a.Len(resp.Boxes, int(maxBoxNumToGet))
+
+	for i := 0; i < len(testingBoxNames); i += 16 {
+		var strSliceTest []string
+		// grouping box names to operate, and delete such boxes
+		if i+16 >= len(testingBoxNames) {
+			strSliceTest = testingBoxNames[i:]
+		} else {
+			strSliceTest = testingBoxNames[i : i+16]
+		}
+		operateAndMatchRes("delete", strSliceTest)
+	}
+
+	resp, err = testClient.ApplicationBoxes(uint64(createdAppID), 0)
+	a.NoError(err)
+	a.Empty(resp.Boxes)
+
+	// Get Box value from box name
+	encodeInt := func(n uint64) []byte {
+		ibytes := make([]byte, 8)
+		binary.BigEndian.PutUint64(ibytes, n)
+		return ibytes
+	}
+
+	boxTests := [][]interface{}{
+		{[]byte("foo"), "str:foo", []byte("bar12")},
+		{encodeInt(12321), "int:12321", []byte{0, 1, 254, 3, 2}},
+		{[]byte{0, 248, 255, 32}, "b64:APj/IA==", []byte("lux56")},
+	}
+	for _, boxTest := range boxTests {
+		boxName := boxTest[0].([]byte)
+		encodedName := boxTest[1].(string)
+		// Box values are 5 bytes, as defined by the test TEAL program.
+		boxValue := boxTest[2].([]byte)
+		operateBoxAndSendTxn("create", []string{string(boxName)}, []string{""})
+		operateBoxAndSendTxn("set", []string{string(boxName)}, []string{string(boxValue)})
+
+		boxResponse, err := testClient.GetApplicationBoxByName(uint64(createdAppID), encodedName)
+		a.NoError(err)
+		a.Equal(boxName, boxResponse.Name)
+		a.Equal(boxValue, boxResponse.Value)
+	}
```

Comment:
The test does a convincing job of testing happy paths. I think it would also benefit to test some error conditions. Ideally, we would also want to test the new box opcodes as well.

Reply:
resolving as #4576 proposes to address